hack/remove-members.go

```diff
@@ -0,0 +1,184 @@
+	"sort"
+	"strings"
+)
+
+var dryrun bool
+var repoRoot string
+
+func main() {
+	flag.StringVar(&repoRoot, "root", ".", "Root of the repository")
+	flag.BoolVar(&dryrun, "dryrun", true, "Enable Dryrun or not")
+
+	flag.Parse()
+
+	if len(flag.Args()) < 1 {
+		fmt.Print("Usage: remove-members [flags] <file-containing-members-list>\n\nFlags:\n\t-root\tstring\tpath to root directory of the repository\n\t-dryrun\tboolean\tEnable/Disable dryrun (Default: Enabled)")
+		os.Exit(0)
+	}
+
+	memberList, err := readMemberList(flag.Args()[0])
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	configPath := repoRoot + "/config"
+
+	err = removeMembers(memberList, configPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+}
+
+//readMemberList reads the list of members to be removed from the given filepath
+func readMemberList(path string) ([]string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	var members []string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		members = append(members, scanner.Text())
+	}
+	return members, scanner.Err()
+}
+
+//removeMembers walks through the config directory and removes the occurences of the given member name
+func removeMembers(memberList []string, configPath string) error {
+	for _, member := range memberList {
+		var orgs, teams []string
+		count := 0
+		fmt.Print(member)
+
+		err := filepath.Walk(configPath, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+
+			if matched, err := filepath.Match("*.yaml", filepath.Base(path)); err != nil {
+				return err
+			} else if matched {
+				removed, err := removeMemberFromFile(member, path)
+				if err != nil {
+					return err
+				}
+
+				//Record the org/team name when a member is removed from it
+				if removed {
+					count++
+					if info.Name() == "org.yaml" {
+						orgs = append(orgs, filepath.Base(filepath.Dir(path)))
+					}
+					if info.Name() == "teams.yaml" {
+						teams = append(teams, filepath.Base(filepath.Dir(path)))
+					}
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		sort.Strings(orgs)
+		sort.Strings(teams)
+
+		fmt.Printf("\n Orgs: %v\n Teams: %v\n Number of occurences: %d\n", orgs, teams, count)
+
+		//Proceed to committing changes if member is actually removed from somewhere
+		if count > 0 {
+			commitRemovedMembers(member, orgs, teams)
+		}
+	}
+
+	return nil
+}
+
+func removeMemberFromFile(member string, path string) (bool, error) {
+
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return false, err
+	}
+
+	re := regexp.MustCompile(`(\s+)?- ` + member + `(.*)?`)
+
+	if re.Match(content) {
+
+		//Mofify the file only if it's not a dry run
+		if dryrun == true {
+			return true, nil
+		}
+
+		updatedContent := re.ReplaceAll(content, []byte(""))
+		err = ioutil.WriteFile(path, updatedContent, 0666)
+
+		if err != nil {
+			return false, err
+		}
```

Comment:
can combine the if condition on line 132

(same changes at other places in this file)